main/2025/CG-02.md

```diff
@@ -130,12 +198,2205 @@ sessions](https://calendar.google.com/calendar/u/0?cid=Y18zYmNkNzdiOWZmZjM4YzA0Y
+
+CW: we might want to think about how we allocate when using data descriptors, since currently we separately copy them in. worth thinking about how this would influence what we do in the future.
+
+DG: yeah just wanted to highlight this as an avenue for feedback.
+
+BV: Are the contents of ImmutableArrayBuffers allowed to change? (If the backing memory changes)
+
+SYG: I think that would be very surprising. Good to clarify with the proposal champions. I would imagine the readonly bit is truly readonly at the OS level. Not read-only by user code only.
+
+DG: There is now immutable view at this time.
+
+SYG: all views on immutable buffers would be presented as immutable. If you query the properties it would show as non-writable.
+
+KM: I haven't looked too closely, but my understanding is that there is no hook to provide for modifying the immutable buffers, so doing so would violate the spec.
+
+SYG: I would object to the proposal in TC39 if it meant “RO by user code only” thats really weird.
+
+RH: My understanding is that if you grab a Wasm Memory's buffer, you cannot detach(?) that buffer.
+
+DG: yes
+
+RH: so transferToImmutable just wouldn’t work. One other thing we could do, if we had something like the static memory protection, You can imagine getting a read-only buffer. You still have to do index translation, but it's still possible.
+
+DG: yeah index translation is just something you have to do. Unpleasant but that’s it. 
+Will discuss detachability in a followup conversation, do we want to make ArrayBuffers backed by wasm memory detachable. Will discuss in the future.
+
+CW: Do you have a one sentence motivation why they are not detachable right now?
+
+DG: wasn’t able to find. They are not detachable out of convenience, but maybe we should move to a different model. 
+
+CW: We maybe have discussed that we don't want Wasm memories to have multiple aliases, but that might have been based on vibes.
+
+DG: yeah it was definitely just a conservative decision
+
+CW: I remember, it was when talking about resizable array buffers, we didn’t want a resizable and nonresizable alias at once.
+
+DG: so it seems we don't have any current feedback for stage 2.7 in TC39, and we might want to expose to wasm in some way in the future.
+
+RR (in chat): Surprised they're not transferrable! We might have issues with moving them between web workers for threading purposes? No need to use immutability tor that, I guess, I'm just baffled by the restriction.
+
+DT (in chat): The discussion has me thinking about Wasm access to mapped webgl/webgpu buffers... has anyone in chat done any work related to this?
+
+DG: please contact us with any feedback on any of these topics
+
+#### [Compilation Hints] (Emanuel Ziegler)
+
+TODO: slides
```

Comment:
@ecmziegler, can you provide a link to your slides?

Reply:
https://docs.google.com/presentation/d/1h_IFj5Y1K2pZq2LwKpDi7egAGPxDL1y4-Z1KGG7RnQo/edit?usp=sharing